src/main/scala/org/broadinstitute/dsde/firecloud/service/OAuthService.scala

```diff
@@ -26,189 +37,120 @@ object OAuthService {
   val remoteTokenDatePath = FireCloudConfig.Rawls.authPrefix + "/user/refreshTokenDate"
   val remoteTokenDateUrl = FireCloudConfig.Rawls.baseUrl + remoteTokenDatePath
 
+  sealed trait OauthServiceMessage
+  case class HandleOauthCode(code: String, redirectUri: String) extends OauthServiceMessage
+  case class GetRefreshTokenStatus(userInfo: UserInfo) extends OauthServiceMessage
+
+  def props(oauthService: () => OAuthService): Props = {
+    Props(oauthService())
+  }
+
+  def constructor(app: Application)()(implicit executionContext: ExecutionContext) =
+    new OAuthService(app.thurloeDAO)
 }
 
-trait OAuthService extends HttpService with PerRequestCreator with FireCloudDirectives with FireCloudRequestBuilding {
+class OAuthService(val thurloeDao: ThurloeDAO)(implicit protected val executionContext: ExecutionContext) extends Actor
```

Comment:
I see a bunch of spray client code in this actor I would expect that to be in DAO classes.